mpqp/core/instruction/measurement/basis_measure.py

```diff
@@ -49,12 +50,14 @@ class BasisMeasure(Measure):
 
     def __init__(
         self,
-        targets: list[int],
+        targets: Optional[list[int]] = None,
         c_targets: Optional[list[int]] = None,
         shots: int = 0,
```

Comment:
I have a suggestion following one of Julien's remark one day. Saying that if I want a StateVector, I can not put a measure in the circuit, or also do BasisMeasure() (with this new feature, which will put shots=0, and measure all the qubits), but then it is really bringing something? 

While if someone pulls out the BasisMeasure, probably it is to specify the targets and still put shots=0, or specify the shots (and possibly the target).

In that sense, I propose that we put the shots default equal to 1024 or something like this, which will permit quckily to have a SAMPLE job for the whole circuit by just putting BasisMeasure().

This will be a bit a breaking change.